src/query/storage/promremote/query_coverter.go

```diff
@@ -30,11 +30,11 @@ import (
 	"github.com/prometheus/prometheus/prompb"
 )
 
-var errNilQuery = errors.New("received nil query")
+var errNilQuery = errors.New("received nil query or no samples in query")
 
 func convertAndEncodeWriteQuery(queries []*storage.WriteQuery) ([]byte, error) {
 	promQuery := convertWriteQuery(queries)
-	if len(promQuery.Timeseries) == 0 {
+	if len(promQuery.Timeseries) == 0 || len(promQuery.Timeseries[0].Samples) == 0 {
```

Comment:
Can `promQuery` have multiple time series?

Reply:
it usually won't have multiple series but it could, let me think about this, i think we could strip time series that has empty samples here

Reply:
Actually this is not the right place, doing it inside `convertWriteQuery`